src/util.cpp

```diff
@@ -630,13 +631,66 @@ boost::filesystem::path GetConfigFile()
     return pathConfigFile;
 }
 
+// This one returns a valid path only if we specify the "-enable_mc_crypto_logger" option
+// otherwise it returns an empty string. This case makes init_zendoo() function forcefully fail,
+// thus disabling the logging function altogether.
 boost::filesystem::path GetMcCryptoConfigFile()
 {
-    boost::filesystem::path pathConfigFile(GetArg("-mc_crypto_conf", "mc_crypto_log_config.yaml"));
-    if (!pathConfigFile.is_absolute())
-        pathConfigFile = GetDataDir(false) / pathConfigFile;
+    if (GetBoolArg("-enable_mc_crypto_logger", false)) 
+        return GetDataDir(false) / "mc_crypto_log_config.yaml";
+    else
+        return boost::filesystem::path("");
+}
 
-    return pathConfigFile;
+void createMcCryptoLogConfigFile() {
+    LogPrintf("mc-crypto logger configuration file not found! Creating a new on in current dataDir\n");
+
+const char* configFilePayload = 
+"appenders:\n\
+\n\
+  file:\n\
+    kind: rolling_file\n\
+    path: FILENAME_PLACEHOLDER\n\
+    encoder:\n\
+      pattern: \"{d(%Y-%m-%d %H:%M:%S)} | {({l}):5.5} | {f}:{L} - {m}{n}\"\n\
+    policy:\n\
+      trigger:\n\
+        kind: size\n\
+        limit: 10 kb\n\
+      roller:\n\
+        kind: fixed_window\n\
+        pattern: FILENAME_PLACEHOLDER{}.gz\n\
```

Comment:
I'd suggest to add one more folder level as default; even if probably it's a matter of taste, I'd prefer default not spreading those files directly in datadir.

```suggestion
        pattern: FILENAME_ARCHIVE_PLACEHOLDER{}.gz\n\
        count: 5\n\
        base: 1\n\
\n\
root:\n\
  level: info\n\
  appenders:\n\
    - file\n";

    // Replace FILENAME_PLACEHOLDER/FILENAME_ARCHIVE_PLACEHOLDER with the proper paths (depending on the OS!)
    boost::filesystem::path logFilename = GetDataDir(false) / "mc_crypto.log";
    boost::filesystem::path logFilenameArchive = GetDataDir(false) / "mc_crypto_logs" / "mc_crypto.log";

    std::string updatedContent = std::regex_replace(logConfigContents, std::regex("FILENAME_PLACEHOLDER"), logFilename.c_str());
    updatedContent = std::regex_replace(updatedContent, std::regex("FILENAME_ARCHIVE_PLACEHOLDER"), logFilenameArchive.c_str());
```

Reply:
Thanks for the comment. I would leave the directories structure as it is, as with the default configuration just an handful of files gets created, and leave the user the task of customizing these aspect by altering the configuration files.